src/me/eglp/gv2/commands/fun/CommandReminder.java

```diff
@@ -0,0 +1,191 @@
+package me.eglp.gv2.commands.fun;
+
+import java.awt.Color;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import me.eglp.gv2.util.GraphiteUtil;
+import me.eglp.gv2.util.base.GraphiteMessageChannel;
+import me.eglp.gv2.util.base.guild.GraphiteGuildMessageChannel;
+import me.eglp.gv2.util.base.guild.GraphiteModule;
+import me.eglp.gv2.util.base.guild.reminder.GuildReminder;
+import me.eglp.gv2.util.base.guild.reminder.ReminderRepetitionEnum;
+import me.eglp.gv2.util.command.Command;
+import me.eglp.gv2.util.command.CommandCategory;
+import me.eglp.gv2.util.command.CommandInvokedEvent;
+import me.eglp.gv2.util.command.ParentCommand;
+import me.eglp.gv2.util.emote.JDAEmote;
+import me.eglp.gv2.util.emote.unicode.GraphiteUnicode;
+import me.eglp.gv2.util.lang.DefaultLocaleString;
+import me.eglp.gv2.util.lang.DefaultMessage;
+import me.eglp.gv2.util.lang.GraphiteTimeParser;
+import me.eglp.gv2.util.lang.defaults.DefaultLocale;
+import me.eglp.gv2.util.lang.defaults.DefaultLocales;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.channel.ChannelType;
+import net.dv8tion.jda.api.entities.emoji.Emoji;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+
+/**
+ * This is the reminder Command class. It handles the different sub-commands and deligates the functionality to {@link GuildReminder}.
+ * 
+ * @author The Arrayser
+ * @date 2023.03
+ */
+
+public class CommandReminder extends ParentCommand {
+	
+	public CommandReminder() {
+		super(GraphiteModule.FUN, CommandCategory.FUN, "reminder");
+		setDescription(DefaultLocaleString.COMMAND_REMINDER_DESCRIPTION);
+		
+		addSubCommand(new Command(this, "create") {
+			
+			@Override
+			public void action(CommandInvokedEvent event) {
+				//"{prefix}reminder create <date and time> <reminder message> [repeat (1y2d)] [channel]"
+				String date = (String) event.getOption("date_and_time");
+				String reminderMessage = (String) event.getOption("reminder_message");
+				
+				// optional
+				String repeat = (String) event.getOption("repeat");
+				ReminderRepetitionEnum repeatE; 
+				if(repeat == null) {
+					repeatE = null;
+				}else {
+					repeatE = ReminderRepetitionEnum.valueOf(repeat);
+				}
+				GraphiteGuildMessageChannel channel = (GraphiteGuildMessageChannel) event.getOption("channel");
+				if(channel == null) {
+					//Use the current channel instead
+					channel = event.getGuildChannel();
+				}
+				
+				LocalDateTime dateMs = null;
+				
+				try {
+					 dateMs =  LocalDateTime.parse(date, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
```

Comment:
I don't know whether `ISO_LOCAL_DATE_TIME` is very user friendly. Maybe a custom time format (using `DateTimeFormatter`) would be better?